main.go

```diff
@@ -331,3 +341,36 @@ func initKubeConfig(kcLocation string) *rest.Config {
 
 	return kubeConfig
 }
+
+func initTransport(upstreamCAFile string) (http.RoundTripper, error) {
+	if upstreamCAFile == "" {
+		return nil, nil
+	}
+
+	rootPEM, err := ioutil.ReadFile(upstreamCAFile)
+	if err != nil {
+		return nil, err
+	}
+
+	roots := x509.NewCertPool()
+	if ok := roots.AppendCertsFromPEM([]byte(rootPEM)); !ok {
+		return nil, err
```

Comment:
this block doesn't have an `err` instance at hand, just an `ok == false`, if any certificates were not parsed successfully. I think this block needs an explicit `return nil, errors.New("...")`.